vulture/core.py

```diff
@@ -68,6 +89,32 @@ class VultureInputException(Exception):
     pass
 
 
+# Reinventing wheel here to keep vulture lightweight.
+def memoize(func):
+    _cache = {}
+
+    @wraps(func)
+    def wrapped(*args):
+        try:
+            return _cache[args]
+        except KeyError:
+            _cache[args] = result = func(*args)
+            return result
+    return wrapped
+
+
+@memoize
+def estimate_lines(node):
+    """
+    Recursively count child AST nodes under `node`. It is an approximation
+    of the amount of code belonging to the node, which is useful for
+    sorting the list of unused code that a developer might want to remove.
+    """
+    return 1 + sum(estimate_lines(child)
+                   for field in TRAVERSABLE_FIELDS.get(node.__class__, ())
+                   for child in getattr(node, field))
+
+
```

Comment:
please move estimate_lines, memoize and TRAVERSABLE_FIELDS to a new module called "lines.py".